### ISSUE TITLE
ldc: error: ‘::isnan’ has not been declared

### ISSUE THREAD

**Issue by timotheecour** [Linuxbrew/homebrew-core#997]
```
brew gist-logs ldc
```

https://gist.github.com/8c938c498c573d0a464fea9984e01058

```
==> Installing ldc
==> Downloading https://github.com/ldc-developers/ldc/releases/download/v1.0.0/ldc-1.0.0-src.tar.gz
Already downloaded: /home/timothee/.cache/Homebrew/ldc-1.0.0.tar.gz
==> Downloading https://github.com/ldc-developers/ldc/releases/download/v0.17.1/ldc-0.17.1-src.tar.gz
Already downloaded: /home/timothee/.cache/Homebrew/ldc--ldc-lts-0.17.1.tar.gz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/katrina/home/timothee/ubuntu1604/brew/Cellar/ldc/1.0.0_1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST
==> make
Last 15 lines from /home/timothee/.cache/Homebrew/Logs/ldc/02.make:
                 from /katrina/home/timothee/ubuntu1604/brew/Cellar/llvm/3.8.1/include/llvm/ADT/ArrayRef.h:13,
                 from /katrina/home/timothee/ubuntu1604/brew/Cellar/llvm/3.8.1/include/llvm/ADT/APInt.h:19,
                 from /katrina/home/timothee/ubuntu1604/brew/Cellar/llvm/3.8.1/include/llvm/ADT/APFloat.h:20,
                 from /tmp/ldc-20161004-47852-1pavfhe/ldc-1.0.0-src/ldc-lts/dmd2/root/port.c:677:
/usr/include/c++/5/cmath:621:5: note:   ‘std::isinf’
     isinf(_Tp __x)
     ^
CMakeFiles/LDCShared.dir/build.make:1444: recipe for target 'CMakeFiles/LDCShared.dir/dmd2/root/port.c.o' failed
make[2]: *** [CMakeFiles/LDCShared.dir/dmd2/root/port.c.o] Error 1
make[2]: Leaving directory '/tmp/ldc-20161004-47852-1pavfhe/ldc-1.0.0-src/ldc-lts/build'
CMakeFiles/Makefile2:182: recipe for target 'CMakeFiles/LDCShared.dir/all' failed
make[1]: *** [CMakeFiles/LDCShared.dir/all] Error 2
make[1]: Leaving directory '/tmp/ldc-20161004-47852-1pavfhe/ldc-1.0.0-src/ldc-lts/build'
Makefile:163: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by rwhogg** [comment]
Thanks a bunch for the bug report @timotheecour. (You've reported a lot of them now!)

The reason this issue doesn't occur on 14.04 is because it's using an older GCC version. In GCC 5.3 (and maybe as early as 5.0, I'm not sure) `::isnan` is not available but `std::isnan` is. Same deal for `::isinf`.

This was reported [here](https://github.com/ldc-developers/ldc/issues/1395) for the 0.17.1 branch, and a fix was provided [here](https://github.com/ldc-developers/ldc/pull/1403).

Unfortunately, as of 1.0.0 (the release currently in Brew) we need a working D compiler to build `ldc` and the newest one available to build with is 0.17.1. Furthermore, we can't use 0.16 according to [this issue](https://github.com/ldc-developers/ldc/issues/1702).

Given that 1.10 is not available yet, I think our best option is to temporarily patch 0.17.1 to compile correctly. @timotheecour, I'd welcome a PR if you'd like to try that out; alternatively, I can do it myself sometime later this week.

**Comment by rwhogg** [comment]
In the longer term, we might also want to ask upstream if they can make a 0.17.2 release which includes the fix.
